### DISCUSSION TITLE
GetStaticPath create heavy static data

### DISCUSSION THREAD

**Discussion by t-hoang-cna** [vercel/next.js#20387] [Help]
Problem:
I have a page that display hotel information by id, so I use getStaticPath to create thing like /hotel-info/542711, 542711 is hotel-id

Problem is There are thousands of hotels, And NextJS will pre-build all that thousands page???? (Increment Static Generation)

Is there problem with memory due to store too much pre-built page like that? ...

**Comment by rafaelalmeidatk** [comment]
If you have a lot of pages to build, you can avoid creating all of them on every build by using `fallback: true` in `getStaticPaths`. Then you can return only the most visited hotels in `getStaticPaths` and the other ones are going to be built during runtime. See more details here: https://nextjs.org/docs/basic-features/data-fetching#fallback-true
